cli/crates/backend/src/api/link.rs

```diff
@@ -0,0 +1,96 @@
+use super::{
+    client::create_client,
+    consts::{API_URL, PROJECT_METADATA_FILE},
+    errors::ApiError,
+    graphql::queries::viewer::{PersonalAccount, Viewer},
+    types::{AccountWithProjects, Project, ProjectMetadata},
+    utils::project_linked,
+};
+use common::environment::Environment;
+use cynic::{http::ReqwestExt, QueryBuilder};
+use std::iter;
+
+/// # Errors
+///
+/// see [`ApiError`]
+#[allow(clippy::module_name_repetitions)]
+pub async fn get_viewer_data_for_link() -> Result<Vec<AccountWithProjects>, ApiError> {
+    // TODO consider if we want to do this elsewhere
+    if project_linked() {
```

Comment:
We seem to be mixing synchronous APIs with async. This checks if a file exists with a synchronous API within an `async` function. I know the CLi is a mix match but whenever we're in an `async` scope already, it'd be desirable for all I/O within to play nicely as well. Or we wrap it with `tokio::task::spawn_blocking()` if there's no async API to do the same thing.

Reply:
I guess all I meant to say is that we should use [`tokio::fs::try_exists()`](https://docs.rs/tokio/latest/tokio/fs/fn.try_exists.html) here. 🙂

Reply:
Makes sense, fixed!